fiat-web/src/main/java/com/netflix/spinnaker/fiat/controllers/AuthorizeController.java

```diff
@@ -307,19 +307,26 @@ public void canCreate(
         if (unrestricted.isPresent()) {
           log.debug(
               "Falling back to unrestricted user permissions for user {}", authenticatedUserId);
-          userPermission = unrestricted.get().setId(authenticatedUserId);
+          userPermission =
+              new UserPermission().setId(authenticatedUserId).merge(unrestricted.get());
         }
       }
+      log.debug(
+          "Returning fallback permissions (user: {}, accounts: {}, roles: {})",
+          userId,
+          (userPermission != null) ? userPermission.getAccounts() : Collections.emptyList(),
+          (userPermission != null)
+              ? userPermission.getRoles().stream().map(Role::getName).collect(Collectors.toList())
+              : Collections.emptyList());
+    } else {
+      log.debug(
```

Comment:
One of the cases where we don't fall back happens during gate's auth flow for X509 - this should be more obvious that occurred (userId != authenticatedUserId)